src/Hedgehog/Gen.fs

```diff
@@ -321,17 +321,25 @@ module Gen =
                 1 + n, map Some g
             ])
 
-    let private atLeast (n : int) (xs : List<'a>) : bool =
-        (List.length xs) >= n
-
     /// Generates a list using a 'Range' to determine the length.
-    let list (range : Range<int>) (g : Gen<'a>) : Gen<List<'a>> =
-        Random.sized (fun size -> random {
-            let! k = Random.integral range
-            let! xs = Random.replicate k (toRandom g)
-            return Shrink.sequenceList xs
-                |> Tree.filter (atLeast (Range.lowerBound size range))
-        })
+    let list (range : Range<int>) (gen : Gen<'a>) : Gen<List<'a>> =
```

Comment:
As I've done here, feel free to rename bad parameter names in your PRs if you're touching a function that has them.